desafio-05/qrwteyrutiyoup/d05.go

```diff
@@ -0,0 +1,364 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"sync"
+)
+
+type area struct {
+	name []byte
+	code []byte
+}
+
+type employee struct {
+	name     []byte
+	surname  []byte
+	salary   float64
+	areaCode []byte
+}
+
+type salaryStats struct {
+	biggest       []*employee
+	lowest        []*employee
+	employeeCount uint32
+	total         float64
+	average       float64
+}
+
+type groupSalaryStats struct {
+	employees []*employee
+	salaries  salaryStats
+}
+
+type groupSalaryStatsMap map[string]*groupSalaryStats
+
+type d05 struct {
+	areas         map[string]*area
+	salaries      salaryStats
+	employeeCount uint32
+
+	salaryByArea    groupSalaryStatsMap
+	salaryBySurname groupSalaryStatsMap
+	employeesByArea map[string][]*employee
+}
+
+const (
+	numberOfBlocksDefault = 16
+)
+
+func (dst *salaryStats) mergeSalaryStats(src *salaryStats) {
+	dst.employeeCount += src.employeeCount
+	dst.total += src.total
+	dst.average = dst.total / float64(dst.employeeCount)
+
+	if len(dst.lowest) == 0 || (len(src.lowest) > 0 && src.lowest[0].salary < dst.lowest[0].salary) {
+		dst.lowest = append([]*employee{}, src.lowest...)
+	} else if (len(dst.lowest) > 0 && len(src.lowest) > 0) && dst.lowest[0].salary == src.lowest[0].salary {
+		dst.lowest = append(dst.lowest, src.lowest...)
+	}
+
+	if len(dst.biggest) == 0 || (len(src.biggest) > 0 && src.biggest[0].salary > dst.biggest[0].salary) {
+		dst.biggest = append([]*employee{}, src.biggest...)
+	} else if (len(dst.biggest) > 0 && len(src.biggest) > 0) && dst.biggest[0].salary == src.biggest[0].salary {
+		dst.biggest = append(dst.biggest, src.biggest...)
+	}
+
+}
+
+func (dst groupSalaryStatsMap) mergeGroupSalary(key string, src *groupSalaryStats) {
+	if s, ok := dst[key]; ok {
+		s.salaries.mergeSalaryStats(&src.salaries)
+		return
+	}
+	dst[key] = src
+}
+
+func (stats *d05) makeStats(wg *sync.WaitGroup, block chan *d05) {
+	for {
+		select {
+		case s := <-block:
+			stats.employeeCount += s.employeeCount
+
+			for k, _ := range s.areas {
```

Comment:
You can use `for k := range s.areas` (no need for the underscore) (and elsewhere)